packages/wallets/glow/src/adapter.ts

```diff
@@ -174,7 +176,7 @@ export class GlowWalletAdapter extends BaseMessageSignerWalletAdapter {
     async sendTransaction(
         transaction: Transaction,
         connection: Connection,
-        options?: SendTransactionOptions
+        options?: SendTransactionOptions & { network: Network }
```

Comment:
Extending these options is no good, unfortunately. However, you can add the option for glow to the constructor of the adapter like this: https://github.com/solana-labs/wallet-adapter/blob/a827686dbc109ef43b5da0031811cb5013aaaac8/packages/wallets/solflare/src/adapter.ts#L29-L31

Store the config in the adapter's state, then pass it to your wallet extension's methods where needed.

Reply:
Hmm, I think that's a bit annoying for developers. They should be able to decide at the time of the sending tx what network it should go to, it doesn't need to be picked when initializing the `wallet`.

Reply:
What's the user story for needing to pick it except when connecting to the wallet? Dapps aren't really built around cluster switching, and if they do, most wallets use this on init.

Reply:
But anyway, I don't want to change the interface for this, especially because we are looking to standardize the API soon (which will reduce the need to support need adapters at all).

Reply:
> What's the user story for needing to pick it except when connecting to the wallet?

It makes sense to switch between `mainnet` / `devnet` if you want to run something as a test on `devnet` before running it on `mainnet`.